### PR TITLE
feat: add cloudwatch support

### PR DESCRIPTION
**Pull request by Kaitou786**

### CODE REVIEW

**Comment by Kaitou786** [comment]
Can be made ready and merged once the fixes PR is merged

**Comment by Kaitou786** [comment]
[[CC] Cloudwatch](https://trello.com/c/kZYkyMCU/77-cc-cloudwatch)

File: checks/cloudwatch.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	for _, alarm := range alarms.CompositeAlarms {
+		if alarm.StateValue == types.StateValueAlarm {
+			message += fmt.Sprintf("alarm '%s': is in %s state Reason: %s ReasonData: %s\n", *alarm.AlarmName, alarm.StateValue, *alarm.StateReason, *alarm.StateReasonData)
+		}
+	}
+	if message != "" {
+		return result.Failf(message).StartTime(start)
+	}
+	return result.StartTime(start)
```
Comment (moshloop) [inline checks/cloudwatch.go:68]:
```suggestion
	return result
```

Start time is added automatically by pkg.Sueccess

File: checks/cloudwatch.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	check := extConfig.(v1.CloudWatchCheck)
+	result := pkg.Success(check)
+	start := time.Now()
+	cfg, err := awsUtil.NewSession(ctx, check.AWSConnection)
+	if err != nil {
+		return result.ErrorMessage(err).StartTime(start)
+	}
+	client := cloudwatch.NewFromConfig(*cfg)
+	maxRecords := int32(100)
+	alarms, err := client.DescribeAlarms(ctx, &cloudwatch.DescribeAlarmsInput{
```
Comment (moshloop) [inline checks/cloudwatch.go:44]: Can we add the alarms as result data as well ?

File: api/v1/checks.go [hunk 3]
```diff
@@ -160,6 +160,34 @@ func (c S3BucketCheck) GetType() string {
 	return "s3Bucket"
 }
 
+type CloudWatchCheck struct {
+	Description   `yaml:",inline" json:",inline"`
```
Comment (moshloop) [inline api/v1/checks.go:164]: Can we add Templateable ?

File: api/v1/checks.go [hunk 4]
```diff
@@ -160,6 +160,34 @@ func (c S3BucketCheck) GetType() string {
 	return "s3Bucket"
 }
 
+type CloudWatchCheck struct {
```
Comment (moshloop) [inline api/v1/checks.go:163]: Can we add a Cloudwatch wrapper class like the others ?

**Pull request merged**